modules/custom/az_paragraphs/src/AZResponsiveBackgroundImageCssHelper.php

```diff
@@ -0,0 +1,150 @@
+use Drupal\Component\Render\FormattableMarkup;
+
+/**
+ * Class AZResponsiveBackgroundImageCssHelper.
+ *
+ * Adds service to form responsive background image CSS.
+ */
+class AZResponsiveBackgroundImageCssHelper {
+
+  /**
+   * The background image css service.
+   *
+   * @var \Drupal\az_paragraphs\AZBackgroundImageCssHelper
+   */
+  protected $backgroundImageCss;
+
+  /**
+   * Constructs an AZResponsiveBackgroundImageCssHelper.
+   *
+   * @param \Drupal\az_paragraphs\AZBackgroundImageCssHelper $backgroundImageCss
+   *   The background image css helper service.
+   */
+  public function __construct(
+    AZBackgroundImageCssHelper $backgroundImageCss
+    ) {
+    $this->backgroundImageCss = $backgroundImageCss;
+  }
+
+  /**
+   * @return array
+   *   An array with all required settings.
+   */
+  public static function defaultSettings() {
+    return [
+      'bg_image_selector' => 'body',
+      'bg_image_color' => '#FFFFFF',
+      'bg_image_x' => 'center',
+      'bg_image_y' => 'center',
+      'bg_image_attachment' => 'scroll',
+      'bg_image_repeat' => 'no-repeat',
+      'bg_image_background_size' => 'cover',
+      'bg_image_gradient' => '',
+      'bg_image_media_query' => 'all',
+      'bg_image_important' => 0,
+      'bg_image_z_index' => 'auto',
+    ];
+  }
+
+  /**
+   * Adds a responsive background image to the page.
+   *
+   * This function uses the css 'background' property.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $image
+   *   The entity to display.
+   * @param array $css_settings
+   *   An array of css settings to use. Possible values are:
+   *      - bg_image_selector: The css selector to use
+   *      - bg_image_color: The background color
+   *      - bg_image_x: The x offset
+   *      - bg_image_y: The y offset
+   *      - bg_image_attachment: The attachment property (scroll or fixed)
+   *      - bg_image_repeat: The repeat settings
+   *      - bg_image_background_size: The background size property if necessary
+   *      - bg_image_gradient: A CSS background gradient
+   *      - bg_image_media_query: Set the media query (all, print or screen)
+   *      - bg_image_important: Add the `!important` property to all background properties.
+   *      - bg_image_z_index: The z-index property if necessary
+   *    Default settings will be used for any values not provided.
+   * @param string $responsive_image_style
+   *   Add responsive image style to the image before applying it to the
+   *   background.
+   *
+   * @return array
+   *   The array containing the CSS.
+   */
+  public function getResponsiveBackgroundImageCss(EntityInterface $image, array $css_settings = [], $responsive_image_style = NULL) {
+
+    // Merge defaults into css_settings array without overriding values.
+    $css_settings += self::defaultSettings();
+
+    $style_elements = [];
+    $css = [];
+    $with_media_query = '';
+    $selector = HTML::getId($css_settings['bg_image_selector']);
```

Comment:
`HTML::getId` might be a bit too strict, we should consider  `HTML::cleanCssIdentifier`  instead. https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Html.php/function/Html%3A%3AcleanCssIdentifier/8.2.x